### PR TITLE
14290 - GlobalKeyCommands causing unnecessary Deck reshuffles and corruption

### PR DESCRIPTION
**Pull request by BrentEaston**

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/counters/Deck.java [hunk 1]
```diff
@@ -882,17 +882,22 @@ public Command shuffle() {
...
+   * If this is an Always shuffle Deck, then shuffle the list of pieces, otherwise just
+   * reverse the list order so that we deal from the top.
+   *
+   * @return List of pieces in Dealable order
+   */
+  public List<GamePiece> getOrderedPieces() {
+    final List<GamePiece> pieces = asList();
     if (!ALWAYS.equals(shuffleOption)) {
-      return new NullCommand();
+      Collections.shuffle(pieces);
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/counters/Deck.java:895]: Should that be `Collections.shuffle(pieces, gameModule.getRNG());`, to ensure we're using the right RNG?

Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/counters/Deck.java]: Indeed.

Comment (jonathanrwatts) [reply vassal-app/src/main/java/VASSAL/counters/Deck.java]: Am I misreading this, or is this code only shuffling if the shuffleOption is NOT "Always"?  Isn't that backwards?

Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/counters/Deck.java]: The dangers of cut & paste. You are 100% correct. Good pick-up :)

**Pull request merged**